src/main/java/net/glowstone/entity/objects/GlowPainting.java

```diff
@@ -0,0 +1,334 @@
+        TITLE_BY_ART.put(Art.CREEBET, "Creebet");
+        TITLE_BY_ART.put(Art.WANDERER, "Wanderer");
+        TITLE_BY_ART.put(Art.GRAHAM, "Graham");
+        TITLE_BY_ART.put(Art.MATCH, "Match");
+        TITLE_BY_ART.put(Art.BUST, "Bust");
+        TITLE_BY_ART.put(Art.STAGE, "Stage");
+        TITLE_BY_ART.put(Art.VOID, "Void");
+        TITLE_BY_ART.put(Art.SKULL_AND_ROSES, "SkullAndRoses");
+        TITLE_BY_ART.put(Art.WITHER, "Wither");
+        TITLE_BY_ART.put(Art.FIGHTERS, "Fighters");
+        TITLE_BY_ART.put(Art.POINTER, "Pointer");
+        TITLE_BY_ART.put(Art.BURNINGSKULL, "BurningSkull");
+        TITLE_BY_ART.put(Art.SKELETON, "Skeleton");
+        TITLE_BY_ART.put(Art.DONKEYKONG, "DonkeyKong");
+        TITLE_BY_ART.put(Art.PIGSCENE, "PigScene");
+
+        TITLE_BY_ART.forEach((art, title) -> ART_BY_TITLE.put(title, art));
+    }
+
+    public GlowPainting(Location center) {
+        this(center, BlockFace.SOUTH);
+    }
+
+    public GlowPainting(Location center, BlockFace facing) {
+        super(center, facing);
+        this.center = center;
+        setArtInternal(DEFAULT_ART);
+    }
+
+    public String getArtTitle() {
+        return TITLE_BY_ART.get(art);
+    }
+
+    public static Art getArtFromTitle(String title) {
+        return ART_BY_TITLE.get(title);
+    }
+
+    @Override
+    public EntityType getType() {
+        return EntityType.PAINTING;
+    }
+
+    @Override
+    public boolean entityInteract(GlowPlayer player, InteractEntityMessage message) {
+        if (message.getAction() == Action.ATTACK.ordinal()) {
+            // TODO: use correct RemoveCause
+            if (EventFactory.callEvent(new HangingBreakEvent(this, RemoveCause.DEFAULT)).isCancelled()) {
+                return false;
+            }
+            if (player.getGameMode() != GameMode.CREATIVE) {
+                world.dropItemNaturally(location, new ItemStack(Material.PAINTING));
+            }
+            remove();
+        }
+        return true;
+    }
+
+    @Override
+    public List<Message> createSpawnMessage() {
+        int x = center.getBlockX();
+        int y = center.getBlockY();
+        int z = center.getBlockZ();
+        String title = getArtTitle();
+
+        return Collections.singletonList(
+            new SpawnPaintingMessage(this.getEntityId(), this.getUniqueId(), title, x, y, z, facing.ordinal())
+        );
+    }
+
+    @Override
+    public Art getArt() {
+        return art;
+    }
+
+    @Override
+    public boolean setArt(Art art) {
+        return this.setArt(art, false);
+    }
+
+    @Override
+    public boolean setArt(Art art, boolean force) {
+        Art oldArt = this.art;
+        setArtInternal(art);
+        updateBoundingBox();
+
+        if (!force && isObstructed()) {
+            setArtInternal(oldArt);
+            updateBoundingBox();
+            return false;
+        }
+
+        respawn();
+
+        return true;
+    }
+
+    private void respawn() {
```

Comment:
Add documentation about this method (sends destroy and then spawn messages).
I would rename this method `refresh` and make it available publicly for plugins wanting to use it.